src/openms/source/FEATUREFINDER/MassTraceDetection.cpp

```diff
@@ -251,27 +254,55 @@ namespace OpenMS
       boost::dynamic_bitset<> peak_visited(total_peak_count);
       Size trace_number(1);
 
-      // check presence of FWHM meta data
+      // check presence of m/z peak FWHM, ion mobility or IM peak FWHM meta array
       int fwhm_meta_idx(-1);
-      Size fwhm_meta_count(0);
-      for (Size i = 0; i < work_exp.size(); ++i)
+      int Ion_Mobility_idx(-1);
+      int IM_fwhm_idx(-1);
+
+      // Identify float meta array indices from first spectrum that contains metadata
+      for (const auto& spec : work_exp)
```

Comment:
-refactor this into a function that returns the indices. 
e.g. having a private static function getIMIndices_(const PeakMap& spectra, ..., Size& fwhm_meta_idx, ...)
-consider adding checks for i.e. if all are present there and throw an exception if they are only partially present (only allow none or all)